core/src/main/java/me/lokka30/treasury/plugin/core/utils/UpdateChecker.java

```diff
@@ -72,65 +129,101 @@ private static void handleVersionCheck(String readString) {
             EconomyAPIVersion latestVersion = EconomyAPIVersion.getCurrentAPIVersion();
 
             handlePluginVersioning(plugin, comparisonResult, latestVersionRead);
```

Comment:
This line is the same as the line 125, why have it 2 times? Is it for redability or something?

Reply:
I see that but 
```Java
EconomyProvider provider = providerEconomyProvider.provide();
EconomyAPIVersion providerVersion = provider.getSupportedAPIVersion();
EconomyAPIVersion latestVersion = EconomyAPIVersion.getCurrentAPIVersion();
```
don't seem to modify the parameters nor do anything else than get the parameters for `handleAPIVersioning` so we are callaing the same method with the same values on the `if` and the `else`

Reply:
this time I am not fully sarcastic but you're blind.

if you know how `if (...) { ... } else { ... }` works (and I can't really see that you know by this comment) you know that only 1 of them would be called at certain state. And I can also see that you haven't checked what the if statement's condition is. 

I am not going to respond to this comment any further, this conversation in my opinion is useless.

Reply:
If in the if and the else the exact same line of code is going to be executed, regardless of if the `providerEconomyProvider` is null or not, with the same parameters, why not put it outside of the if?